pkg/operator/metricshandler/fsync_handler.go

```diff
@@ -0,0 +1,42 @@
+package metricshandler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+func EtcdFSyncHandler(ctx context.Context, recorder events.Recorder, client prometheusv1.API) error {
+	result, warns, err := client.QueryRange(ctx, "histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[5m]))", prometheusv1.Range{
+		Start: time.Now(),
+		End:   time.Now(),
+		Step:  1 * time.Second,
+	})
+	if err != nil {
+		return err
+	}
+	if len(warns) > 0 {
+		klog.Warningf("Prometheus warnings received: %#v", warns)
+	}
+	matrix, ok := result.(model.Matrix)
+	if !ok {
+		return fmt.Errorf("result is not Matrix")
+	}
+	for _, s := range matrix {
+		// TODO: Do something useful with these values
```

Comment:
@hexfusion @deads2k these are the numbers I got in AWS, we should start thinking what are the tolerable numbers per provider (and I need to wire getting the provider into this handler, so probably it will turn into a struct or something)

Reply:
actually, I found `cluster_infrastructure_provider` so we can query provider via prometheus :D